src/graph.rs

```diff
@@ -401,3 +378,31 @@ impl Hashes {
         hash != last_hash
     }
 }
+
+#[test]
+fn stat_mtime_resolution() {
+    let temp_dir = tempfile::tempdir().unwrap();
+    let filename = temp_dir.path().join("dummy");
+    let filename = filename.to_str().unwrap();
+
+    // Write once and stat.
+    std::fs::write(filename, "foo").unwrap();
+    let mtime1 = match stat(filename).unwrap() {
+        MTime::Stamp(mtime) => mtime,
+        _ => panic!("File not found: {}", filename),
+    };
+
+    // Sleep for a short interval.
+    std::thread::sleep(std::time::Duration::from_millis(50));
```

Comment:
0.1 microseconds should be enough, no? Windows has 100 nanosecond resolution, Linux and macOS have nanosecond resolution i think.

(Maybe sleeping a tad longer is fine, but 50000x longer seems excessive.)

Reply:
I don't think you can reliably sleep a thread for 0.1us. At least not on Windows, where the resolution is typically on the order of 16ms.

Reply:
But the sleep arg is a min time to sleep, i.e. the scheduler will round up, not down, right? So wouldn't this say "sleep for the minimum quantum" (?)

Reply:
> But the sleep arg is a min time to sleep, i.e. the scheduler will round up, not down, right? So wouldn't this say "sleep for the minimum quantum" (?)

The assumption you're making is that 1us or whatever actually gets passed to the scheduler.
But the windows `Sleep()` API takes an argument in ms. So how certain are you that Rust's stdlib won't round 0.1us down to 0ms.

Anyway I changed the timeout to 1ms and hereby reject any responsibility for any flakiness of this test in the future.

Reply:
> Anyway I changed the timeout to 1ms.

Ok that didn't work on Linux: https://github.com/evmar/n2/runs/5725792854?check_suite_focus=true#step:6:45